### PR TITLE
Rework op_construct.

### PR DESCRIPTION
**Pull request by zherczeg**
The new form should be more suitable for ES5 class implementation. No binary / perf / mem change.

### CODE REVIEW

File: jerry-core/ecma/operations/ecma-function-object.c [hunk 1]
```diff
@@ -874,13 +762,117 @@ ecma_op_function_construct (ecma_object_t *func_obj_p, /**< Function object */
...
+
+    ecma_free_value (prototype_prop_value);
+
+    this_arg_value = ecma_make_object_value (new_this_obj_p);
+  }
+
+  /* 8. */
+  ecma_value_t ret_value;
+
+  switch (type)
```
Comment (LaszloLango) [inline jerry-core/ecma/operations/ecma-function-object.c]: An `if-else` statement fits better here than a `switch`

Comment (rerobika) [reply jerry-core/ecma/operations/ecma-function-object.c]: Currently it fits better but for further [cases](https://github.com/jerryscript-project/jerryscript/pull/2439/files#diff-c790ca6e2b99859fd40e7cce2885c2fbR1068) switch is more suitable.

Comment (LaszloLango) [reply jerry-core/ecma/operations/ecma-function-object.c]: OK, leave as it is.

**Comment by LaszloLango** [comment]
LGTM

**Pull request merged**